app/uk/gov/hmrc/customs/declarations/information/connectors/ApiSubscriptionFieldsConnector.scala

```diff
@@ -32,25 +36,26 @@ class ApiSubscriptionFieldsConnector @Inject()(http: HttpClient,
                                                config: InformationConfigService)
                                               (implicit ec: ExecutionContext) {
 
-  def getSubscriptionFields[A](apiSubsKey: ApiSubscriptionKey)(implicit hci: HasConversationId, hc: HeaderCarrier): Future[ApiSubscriptionFieldsResponse] = {
-    val url = ApiSubscriptionFieldsPath.url(config.informationConfig.apiSubscriptionFieldsBaseUrl, apiSubsKey)
-    get(url)
-  }
-
-  private def get[A](url: String)(implicit hci: HasConversationId, hc: HeaderCarrier): Future[ApiSubscriptionFieldsResponse] = {
+  def getSubscriptionFields(apiSubsKey: ApiSubscriptionKey)
+                           (implicit hci: HasConversationId, hc: HeaderCarrier): Future[Option[ApiSubscriptionFieldsResponse]] = {
+    val url = new URL(ApiSubscriptionFieldsPath.url(config.informationConfig.apiSubscriptionFieldsBaseUrl, apiSubsKey))
     logger.debug(s"Getting fields id from api subscription fields service. url=$url")
```

Comment:
Could [] be put around the log variables?

Reply:
Oops, missed that one. Fixed.